### DISCUSSION TITLE
Help in decoding the shaft_si excel file

### DISCUSSION THREAD

**Discussion by abhishek-sardar** [petrobras/ross#897] [Q&A]
Hi,

First of all, I want to thank all people involved in building up this fantastic open source tool for rotor dynamics. I have been having trouble deciphering some of the columns in the 'shaft_si' column and would like some help in that.

What do columns 'Element', 'Sub-element', 'Layer' mean in the Rotor Model Data Entry : Multiple Shafts, Elements, Sub-Elements and Layers. Also, what do Station numbers mean i.e. Left station no and Right station no? What does the entry 'layer' mean? I've marked the columns in red.

![Ross](https://user-images.githubusercontent.com/110981964/183897491-880f147e-ad5e-40e0-a62f-c431c1524450.png)

**Comment by raphaeltimbo** [comment]
Hi @abhishek-sardar.
Sorry for the late reply.
Element is the element number. Sometimes you will find repeated values. This happens when you have elements of different diameters which share the same nodes (the term used in the sheet 'stations' is the same as nodes). In the tutorial example we have a centrifugal compressor and the dry gas seal parts are added as additional elements on top of the shaft with this result:
![image](https://user-images.githubusercontent.com/18506378/184946413-2650699d-1c51-4d8e-808a-6c8038e8e22a.png)
In the excel sheet the elements with the same numbers are placed in different 'layers'. 
ROSS does not actually uses the sub-element column.

Parts of this sheet are not used by ROSS, but we kept this format to be able to import files that are used by a different software.

**Reply by abhishek-sardar** [reply]
Thanks Raphael for the explanation. As you said, station is same as node. So, what do left and right station mean? Left is starting from the left end of the rotor and right from the right end of the rotor? (Doesn't seems to be the right explanation?)

Also, in the tutorial, the first step is to create and save material. Instead of that, can we just update the material properties in the excel itself (shaft_si) or creating and saving the material is to be done regardless.

<img width="306" alt="Ross-1" src="https://user-images.githubusercontent.com/110981964/185056282-b49caf2b-dd32-44ff-8de7-059d9aaf8982.png">

**Reply by raphaeltimbo** [reply]
> Left is starting from the left end of the rotor and right from the right end of the rotor?

Yes. There is just one point to watch out: in the spreadsheet the stations have an index that starts at 1, and when loaded with ROSS this station will be 0. So the first element in the spreadsheet will have left station 1 and right station 2, and inside ROSS this element will have left station 0 and right station 1.

> Also, in the tutorial, the first step is to create and save material. Instead of that, can we just update the material properties in the excel itself (shaft_si) or creating and saving the material is to be done regardless.

You don't need to create and save the material if you are loading from the excel file.
The following code for example will run fine:

```python
import ross as rs
from pathlib import Path

shaft_file = Path(rs.__file__).parent / "tests/data/shaft_si.xls"
bearing_file = Path(rs.__file__).parent / "tests/data/bearing_seal_si.xls"
shaft = rs.ShaftElement.from_table(
    file=shaft_file, sheet_type="Model", sheet_name="Model"
)

disks = rs.DiskElement.from_table(file=shaft_file, sheet_name="More")

bearing1 = rs.BearingElement.from_table(n=7, file=bearing_file)
bearing2 = rs.BearingElement.from_table(n=48, file=bearing_file)

bearings = [bearing1, bearing2]

rotor = rs.Rotor(shaft, disks, bearings)

rotor.plot_rotor()
````

**Reply by abhishek-sardar** [reply]
Thanks Raphael!
